tests/integration/api/mutations/removeUserPermissions/removeUserPermissions.test.js

```diff
@@ -0,0 +1,184 @@
+  await testApp.collections.Accounts.updateOne({ _id: mockOtherAccount._id }, {
+    $set: {
+      groups: []
+    }
+  });
+
+  await testApp.collections.users.updateOne({ _id: mockOtherAccount._id }, {
+    $set: {
+      roles: {}
+    }
+  });
+});
+
+test("anyone can with the required permissions can remove a group(s) to an account", async () => {
+  await testApp.setLoggedInUser(mockAdminAccount);
+
+  const groupName = "test-group-1";
+  // The group that is to be removed from the  account
+  const testGroup1 = Factory.Group.makeOne({
+    createdBy: null,
+    name: groupName,
+    permissions: ["test-group-1-permission"],
+    slug: groupName,
+    shopId
+  });
+
+  await testApp.collections.Groups.insertOne(testGroup1);
+
+  const accountBeforeGroupRemoval = await testApp.collections.Accounts.findOneAndUpdate(
+    {
+      _id: mockOtherAccountId
+    }, {
+      $addToSet: {
+        groups: {
+          $each: [groupName]
+        }
+      }
+    },
+    { returnOriginal: false }
+  );
+
+
+  await removeUserPermissions({ groups: ["test-group-1"], shopId: shopOpaqueId, accountId: mockOtherAccountIdOpaque });
+  const accountAfterGroupRemoval = await testApp.collections.Accounts.findOne({ _id: mockOtherAccount._id });
+
+  expect(accountBeforeGroupRemoval.value.groups.length).toEqual(1);
+  expect(accountBeforeGroupRemoval.value.groups).toEqual(expect.arrayContaining(["test-group-1"]));
+  expect(accountAfterGroupRemoval.groups).toEqual(expect.not.arrayContaining(["test-group-1"]));
+  expect(accountAfterGroupRemoval.groups.length).toBe(0);
+});
+
+
+test("anyone without the required permissions should be denied access to remove group from an account", async () => {
+  await testApp.setLoggedInUser(mockOtherAccount);
+
+  const groupName = "test-group-1";
+  // The group that is to be added tp th  account
+  const testGroup1 = Factory.Group.makeOne({
+    createdBy: null,
+    name: groupName,
+    permissions: ["test-group-1-permission"],
+    slug: groupName,
+    shopId
+  });
+
+  await testApp.collections.Groups.insertOne(testGroup1);
+
+  await testApp.collections.Accounts.findOneAndUpdate(
+    {
+      _id: mockAdminAccount._id
+    }, {
+      $addToSet: {
+        groups: {
+          $each: [groupName]
+        }
+      }
+    },
+    { returnOriginal: false }
+  );
+
+  let err = null;
+  try {
+    await removeUserPermissions({ groups: ["test-group-1"], shopId: shopOpaqueId, accountId: mockAdminAccountIdOpaque });
+  } catch (errors) {
+    err = errors;
+  }
+  expect(err).toBeTruthy();
+  expect(err[0]).toMatchSnapshot();
+});
+
+
+test("should throw if a groups array is empty", async () => {
+  await testApp.setLoggedInUser(mockAdminAccount);
+
+  let err = null;
+  try {
+    await removeUserPermissions({ groups: [], shopId: shopOpaqueId, accountId: mockAdminAccountIdOpaque });
+  } catch (errors) {
+    err = errors;
+  }
+  expect(err).toBeTruthy();
+  expect(err[0]).toMatchSnapshot();
+});
```

Comment:
Should add a test, removing permissions from an account, then trying a query or mutation that requires the permissions that were just removed.